File: PHP/Medium/Sorting/3016__Minimum_Number_of_Pushes_to_Type_Word_II.php
<?php

function minimumPushes(string $word): int
{
    $freq = array_count_values(str_split($word));
    arsort($freq);
    $sort_freq = array_values($freq);

    $total_pushes = 0;
    foreach ($sort_freq as $i => $count) {
        $pushes = (int)($i / 8) + 1;
        $total_pushes += $count * $pushes;
    }
    return $total_pushes;
}

$word = "abcde";
echo minimumPushes($word);

File: Python/Medium/Sorting/3016__Minimum_Number_of_Pushes_to_Type_Word_II.py
from collections import Counter


def minimumPushes(word: str) -> int:
    frequency = Counter(word)
    sorted_freq = sorted(frequency.values(), reverse=True)
    total_pushes = 0
    for i, count in enumerate(sorted_freq):
        pushes = (i // 8) + 1
        total_pushes += pushes * count
    return total_pushes


word = "aabbccddeeffgghhiiiiii"

print(minimumPushes(word))
